### PR TITLE
I18n setup

### PR DESCRIPTION
**Pull request by eddiekeller**
*Issue #, if available:*

n/a

*Description of changes:*

adding initial setup for i18n support. only added translated text to the `SignIn` page for now until we decide that this is a good path forward. screenshots showing japanese and french - 

Screen Shot 2021-08-17 at 1 36 33 PM
Screen Shot 2021-08-17 at 1 36 59 PM

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/react/src/components/Authenticator/SignIn/SignIn.tsx
Comment (ericclemmons) [inline packages/react/src/components/Authenticator/SignIn/SignIn.tsx:54]: This is another example where having the I18n + default text makes it easier to correlate what we see on the page with where it comes from in the code:

```suggestion
      <Heading level={1}>{I18n.get('Sign in to your account')}</Heading>
```

Real-world anecdote: when we ran into I18n issues last year, a customer would say "Resend Code isn't translated" (or something to that affect).  We'd open up the project, search "Resend Code", and immediately see "Ah! This isn't wrapped with `I18n`!"

It seems small, but keeping context can make these fixes into a 1-liner and find/replace-able.

File: packages/react/src/components/Authenticator/shared/UserNameAlias.tsx
Comment (ericclemmons) [inline packages/react/src/components/Authenticator/shared/UserNameAlias.tsx:2]: Somebody did _Organize Imports_ ;)

File: examples/next/pages/ui/components/authenticator/sign-in-federated/index.page.tsx
Comment (ericclemmons) [inline examples/next/pages/ui/components/authenticator/sign-in-federated/index.page.tsx:13]: FYI, I found this:
> https://github.com/cypress-io/cypress/issues/7890#issuecomment-824676390

I don't know the best way of testing this, but we could consider having an `components/authenticator/localization/en-GB.page.tsx`, `index.ja-JP.page.tsx`, etc. to test some of this behavior.

Not that we do it here, but it's probably something we'll want to document and support via a `.feature`.

File: packages/react/index.tsx
Comment (ericclemmons) [inline packages/react/index.tsx:6]: We'll need to find another place for this that's not a side-effect.  Unfortunately, that may mean Vue's `setup()` function, React's `Authenticator` with `useEffect`, etc.

I personally had to deal with Angular 9's tree-shaking behavior with side-effects like these that were getting stripped in production builds (but there in development).

Amplify JS lost some bundle optimizations as a result:

- https://webpack.js.org/guides/tree-shaking/#mark-the-file-as-side-effect-free
- https://github.com/aws-amplify/amplify-js/pull/6867
- https://github.com/aws-amplify/amplify-js/pull/6844

Another thing we can do is, rather than `I18n.putVocabularies(dict)` by default, we can instruct customers in our docs to do something like:

> https://quip-amazon.com/UtYxAfbJ7zwz/Intern-Project-Report-Amplify-Issue-Librarian

```js
import de from "@aws-amplify/ui-react/locales/de/translations.json"

I18n.putVocabularies({ de })
```

or

```js
import { de } from "@aws-amplify/ui-react/translations"

I18n.putVocabularies({ de })
```

I'm not sure which is better (the latter seems "cleaner" to me)

**Review by ericclemmons — COMMENTED** [review]
@eddiekeller you've done it again!

This is looking great!  Some small nits that I'd like your opinion on, but the side-effects usage is the one change that we should consider before merging.

File: packages/react/src/components/Authenticator/shared/UserNameAlias.tsx
Comment (eddiekeller) [reply packages/react/src/components/Authenticator/shared/UserNameAlias.tsx:2]: Unorganized imports give me gas :)

File: examples/next/pages/ui/components/authenticator/sign-in-federated/index.page.tsx
Comment (eddiekeller) [reply examples/next/pages/ui/components/authenticator/sign-in-federated/index.page.tsx]: Yeah, I'm changing this back to english. But I'll take a look at adding some tests in in the next PR once this one get approved. The next PR will have me going through all of the React files and adding translations.

File: packages/react/index.tsx
Comment (eddiekeller) [reply packages/react/index.tsx:6]: Oh that's a good idea. Leave it up to the customers. I like that more. Will remove the `.putVocabularies` for now.

File: packages/react/src/components/Authenticator/SignIn/SignIn.tsx
Comment (eddiekeller) [reply packages/react/src/components/Authenticator/SignIn/SignIn.tsx:54]: Ok, I'm fine if we want to do it like that. This pattern that I have in the PR is how we handled it in SageMaker and felt like it was cleaner. But I'm down to forego the `translations` object and just manually wrap each one right inside the HTML if we feel like it would help make things easier to debug.

File: packages/react/src/components/Authenticator/FederatedSignIn/FederatedSignIn.tsx
Comment (eddiekeller) [reply packages/react/src/components/Authenticator/FederatedSignIn/FederatedSignIn.tsx:26]: Same comment as below!

File: packages/react/src/components/Authenticator/SignIn/SignIn.tsx
Comment (ericclemmons) [reply packages/react/src/components/Authenticator/SignIn/SignIn.tsx]: I appreciate that.  We went further than this last year (by even moving the object to yet another file) and when we removed code, we had no way of knowing how much (if any) of that abstracted object was still in use.

So that's where my mind is today – in 6 months, how can I find the relevant code ASAP and, if we delete some of this code, will I prevent unused code? (TypeScript definitions & CSS are great examples of de-coupled code becoming unused)

**Review by wlee221 — APPROVED** [review]
All lgtm 🚢 Just got some general nits, but nothing worth blocking over. Thanks @eddiekeller!

File: packages/core/src/i18n/translations.ts
Comment (eddiekeller) [reply packages/core/src/i18n/translations.ts:2]: copy and pasted from the old copyright in the JS repo. not sure if we need it. None of the other files in this new repo have it.

File: packages/core/src/i18n/translations.ts
Comment (eddiekeller) [reply packages/core/src/i18n/translations.ts:23]: I haven't looked at all the screens yet - just sign in for now. But you're right! I'm sure we will need to add some / modify some eventually.

File: packages/core/src/i18n/translations.ts
Comment (eddiekeller) [reply packages/core/src/i18n/translations.ts:137]: We can add in typing...but I wonder if "forcing" parity between the languages would work. We are going to rely on the open source community for translation support, and I'm picturing a scenario when a contributor will want to add a new string, or only a couple strings of translations. In that case, forcing parity between all languages might almost act like a blocker. Maybe I'm thinking about it wrong.

File: packages/react/src/components/Authenticator/SignIn/SignIn.tsx
Comment (eddiekeller) [reply packages/react/src/components/Authenticator/SignIn/SignIn.tsx:66]: I think translations actually make more sense. Example - Japanese doesn't use spaces. Forcing spaces in the UI would be breaking certain languages. If the intent of the space isn't an actual linguistic space but whitespace between elements, then we should be handling that via CSS instead.

File: packages/core/src/i18n/translations.ts
Comment (wlee221) [reply packages/core/src/i18n/translations.ts:137]: Ah, that's a fair point. I agree, let's keep it as is now. Maybe we can do Partial<> just so that we have vscode autocomplete, but this is not an issue now and I don't want it to be a blocker.

File: packages/react/src/components/Authenticator/SignIn/SignIn.tsx
Comment (wlee221) [reply packages/react/src/components/Authenticator/SignIn/SignIn.tsx:66]: Yeah, but I think with that component the space was used to put spacing between "Forgot your password" and the link if I remember correctly. Which I think should be handled through css or `{{' '}}`

**Pull request merged**
